packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -105,10 +105,10 @@ class CameraValue {
   final FlashMode flashMode;
```

Comment:
I'm curious, if below variables can be null when the value is not initialized (just guessing), why this can be non-null? Also can we improve the nullability in other way to make less nullable variable in the `CameraValue`?

Reply:
Looking at the native code we could indeed initialise the `focusMode` and `exposureMode` fields to `auto` (as this is the default values used on iOS and Android if these are not set). I will update these values. 

The `lockedCaptureOrientation` parameter uses `null` to indicate that no orientation lock is set, meaning it can be set (and default is set) to `null` which should not make it required. The same goes for `recordingOrientation` which is only set during video recording.

Lastly we have the `previewSize` and `errorDescription` parameters which are allowed to be `null` since they are only set once the a camera has been successfully initialised.

Reply:
> Lastly we have the `previewSize` and `errorDescription` parameters which are allowed to be `null` since they are only set once the a camera has been successfully initialised.

How about `late final NonNullType` ?

Reply:
Not sure that would work, my understanding is that `late` should only be used in cases where we can ensure variables are initialised before they are used. In this case both fields can be accessed (or used) before they are initialised (for example when you have instantiated the `CameraController` but haven't called `initialize` (and awaited) yet. Accessing these fields before they are initialised would result in a `LateInitializationError `which might introduce bugs. What do you think?

Reply:
> `late` should only be used in cases where we can ensure variables are initialised before they are used

Yes, that situation is quite similar to the `CameraValue`, while nothing is valid for read or use before it gets initialized.

Here's my common usage with the camera:
* Initialize a controller, and wait til it's initialized.
* After `isInitialized`, call another layout or value binding widgets for use.

In this case, nothing will be reached before initialized, so no exception will be thrown.

Reply:
But the problem here is that `CameraValue` is an immutable class. This would mean that once the class is constructed, it shouldn't change. This means we need to be able to set the `previewSize` field to some default value (which currently is `null`) when it is first constructed. Then when the `CameraController` is initialised a new instance of the `CameraValue` class is created and a new value is assigned to the `previewSize` field.

If we want to convert the `previewSize` field to a `late` field we need to make it private (otherwise people can set it from outside, which is something we don't want being an immutable class), and convert the `final Size previewSize;` field into a getter: `Size get previewSize => _previewSize;` 

Now we would need multiple constructors where one doesn't set the `previewSize` field (this would be the `uninitialized()` named constructor) and the other "normal" constructor does initialise it. This would mean some code duplication. And another side effect would be that we now need to get rid of the `const` keyword for the constructors (which isn't really a big problem in my opinion but still).

Reply:
Agreed. We can ship this and see if any good ideas will be pop up from the community.